src/pages/waiting-room-to-car/components/instructor-registration/instructor-registration.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, Output, EventEmitter, OnChanges } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'instructor-registration',
+  templateUrl: 'instructor-registration.html',
+})
+export class InstructorRegistrationComponent implements OnChanges {
+
+  @Input()
+  instructorRegistration: number;
+
+  @Input()
+  formGroup: FormGroup;
+
+  @Output()
+  instructorRegistrationChange = new EventEmitter<number>();
+
+  private formControl: FormControl;
+
+  ngOnChanges(): void {
+    if (!this.formControl) {
+      this.formControl = new FormControl(null);
+      this.formGroup.addControl('instructorRegistration', this.formControl);
+    }
+    this.formControl.patchValue(this.instructorRegistration);
+  }
+
+  instructorRegistrationChanged(instructorRegistration: string): void {
+    if (this.isNumeric(instructorRegistration)) {
+      this.instructorRegistrationChange.emit(Number.parseInt(instructorRegistration, 10));
+    }
+  }
+
+  isNumeric(value: string): boolean {
```

Comment:
We have this non-trivial function in two places, probably worth extracting.

Reply:
extracted in to a static function on a helper class - will look to see if typescript has extension methods like c# - probably wont yet.